skip triton template precompilation in 311.0-3.11.7 to workaround 311 cpython bug

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #125446
* #125289

Fix for https://github.com/pytorch/pytorch/issues/125374. We dont have CI for this specific versions, but I verified locally. THere is a cpython bug from 3.11.0->3.11.7 where the ast parsing state is global, and errors with multiple threads. when dust settles a little around the new process based compilation we can look into migrating.  


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

eellison committed:
skip precompilation in 311.0-3.11.7 to workaround 311 cpython bug

[ghstack-poisoned]

eellison committed:
Update on "skip triton template precompilation in 311.0-3.11.7 to workaround 311 cpython bug"


Fix for https://github.com/pytorch/pytorch/issues/125374. We dont have CI for this specific versions, but I verified locally. THere is a cpython bug from 3.11->3.17 where the ast parsing state is global, and errors with multiple threads. when dust settles a little around the new process based compilation we can look into migrating. 


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

Chillee commented on torch/_inductor/select_algorithm.py:
```diff
@@ -64,6 +64,14 @@ class KernelNamespace:
 
+@functools.lru_cache(None)
+def warn_ast_parse_3_11():
+    log.warning(
```
Do we actually need a warning here? It doesn't matter that much right - only some compilation time improvement? In that case, I think most users wouldn't care.

eellison replied:
Now that we have a ton more mm configs, it actually matters a good amount.

Chillee replied:
I'd lean towards not having it (i think warning spew is really bad), but if you really want a warning, at least mention what's going to happen if they don't upgrade.

eellison replied:
Yea, fair. I'll remove and then after this is landed try to add a workaround in triton.

eellison committed:
Update on "skip triton template precompilation in 311.0-3.11.7 to workaround 311 cpython bug"


Fix for https://github.com/pytorch/pytorch/issues/125374. We dont have CI for this specific versions, but I verified locally. THere is a cpython bug from 3.11.0->3.11.7 where the ast parsing state is global, and errors with multiple threads. when dust settles a little around the new process based compilation we can look into migrating.  


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

eellison commented:
@pytorchbot merge
